grype/db/v5/namespace/index.go

```diff
@@ -118,6 +118,15 @@ func (i *Index) NamespacesForDistro(d *grypeDistro.Distro) []*distro.Namespace {
 		}
 	}
 
+	// Fall back to alpine:edge if no version segments found
+	// alpine:edge is labeled as alpine-x.x_alphaYYYYMMDD
+	if versionSegments == nil && d.Type.String() == "alpine" {
```

Comment:
I think there is a `grypeDistro.Alpine` to compare against instead of the string (or does that not work in this case?)

Reply:
Re: checking the version, my suggestion here would be to find a way to test this value to see if it matches the `x.y.z` format mentioned in [this comment](https://github.com/anchore/grype/issues/964#issuecomment-1290888755). The logic right now is still just a bit off, because there can be cases where the version is something like `1.2.3-beta`, so `versionSegments` would not be `nil`, and this code won't use edge when it actually should use edge.

Reply:
Updated and added the case for the distro to match the version against a valid regex for `x.y.z` falling back to edge if `x.y.z` is unmatched.